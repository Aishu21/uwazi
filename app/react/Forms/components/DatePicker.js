import React, {Component, PropTypes} from 'react';
import {createFieldClass, controls} from 'react-redux-form';
import DatePickerComponent from 'react-datepicker';
import 'react-datepicker/dist/react-datepicker.css';
import moment from 'moment';

export class DatePicker extends Component {

  constructor(props) {
    super(props);
    this.state = {};
    if (props.value) {
      this.state.value = moment.utc(props.value, 'X');
    }
  }

  onChange(value) {
    if (this.props.endOfDay) {
      value.utc().endOf('day');
    }
    this.props.onChange(value.format('X'));
  }

  componentWillReceiveProps(newProps) {
    if (newProps.value) {
      this.setState({value: moment.utc(newProps.value, 'X')});
    }
  }

  render() {
    return (
        <DatePickerComponent
          dateFormat="MMM DD, YYYY"
          className="form-control"
          onChange={this.onChange.bind(this)}
          selected={this.state.value}
          locale='en-gb'
          fixedHeight
          showYearDropdown
        />
    );
  }

}

DatePicker.propTypes = {
  onChange: PropTypes.func,
  value: PropTypes.string,
  endOfDay: PropTypes.bool
};

export default DatePicker;

const DatePickerField = createFieldClass({
  DatePicker: controls.text
});

export {DatePickerField};
