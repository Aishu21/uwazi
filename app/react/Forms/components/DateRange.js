import React, {Component, PropTypes} from 'react';

import DatePicker, {DatePickerField} from './DatePicker';

export class DateRange extends Component {

  render() {
    return (
        <div>
          <span>From:&nbsp;</span>
          <DatePickerField model={this.props.fromModel} >
            <DatePicker onChange={this.props.fromChange}/>
          </DatePickerField>
          <span>&nbsp;To:&nbsp;</span>
          <DatePickerField model={this.props.toModel} >
            <DatePicker endOfDay={true} onChange={this.props.toChange}/>
          </DatePickerField>
        </div>
    );
  }

}

DateRange.propTypes = {
  fromModel: PropTypes.string,
  toModel: PropTypes.string,
  fromChange: PropTypes.func,
  toChange: PropTypes.func
};

export default DateRange;
