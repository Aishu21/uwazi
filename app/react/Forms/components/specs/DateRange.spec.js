import React from 'react';
import {shallow} from 'enzyme';

import DateRange from '../DateRange';
import DatePicker, {DatePickerField} from '../DatePicker';

describe('DateRange', () => {
  let component;
  let props;

  beforeEach(() => {
    props = {
      fromModel: 'model.from',
      toModel: 'model.to',
      fromChange: jasmine.createSpy('fromChange'),
      toChange: jasmine.createSpy('toChange')
    };
  });

  let render = () => {
    component = shallow(<DateRange {...props}/>);
  };

  it('should render two DatePickerField components passing the correct models', () => {
    render();
    expect(component.find(DatePickerField).first().props().model).toBe('model.from');
    expect(component.find(DatePickerField).last().props().model).toBe('model.to');
  });

  describe('when a date is selected', () => {
    it('should triger onChange events', () => {
      render();
      component.find(DatePicker).first().simulate('change', '1469656800');
      expect(props.fromChange).toHaveBeenCalled();
      component.find(DatePicker).last().simulate('change', '1469656800');
      expect(props.toChange).toHaveBeenCalled();
    });
  });
});
