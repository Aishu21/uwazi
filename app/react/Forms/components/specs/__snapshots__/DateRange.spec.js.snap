// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateRange should allow using the local timezone 1`] = `
<div>
  <div
    className="DatePicker__From"
  >
    <span>
      From:
       
    </span>
    <DatePicker
      endOfDay={false}
      format="YYYY/MM/DD"
      locale="en"
      onChange={[Function]}
      useTimezone={true}
      value={0}
    />
  </div>
  <div
    className="DatePicker__To"
  >
    <span>
       
      To:
       
    </span>
    <DatePicker
      endOfDay={true}
      format="YYYY/MM/DD"
      locale="en"
      onChange={[Function]}
      useTimezone={true}
      value={1}
    />
  </div>
</div>
`;

exports[`DateRange should render the component 1`] = `
<div>
  <div
    className="DatePicker__From"
  >
    <span>
      From:
       
    </span>
    <DatePicker
      endOfDay={false}
      format="YYYY/MM/DD"
      locale="en"
      onChange={[Function]}
      useTimezone={false}
      value={0}
    />
  </div>
  <div
    className="DatePicker__To"
  >
    <span>
       
      To:
       
    </span>
    <DatePicker
      endOfDay={true}
      format="YYYY/MM/DD"
      locale="en"
      onChange={[Function]}
      useTimezone={false}
      value={1}
    />
  </div>
</div>
`;
